src/Tools/BuildValidator/PEReaderExtensions.cs

```diff
@@ -0,0 +1,133 @@
+            exportTableHeader.ReadUInt16();
+            // +0x0C: DLL name RVA
+            exportTableHeader.ReadUInt32();
+            // +0x10: ordinal base
+            int minOrdinal = exportTableHeader.ReadInt32();
+            // +0x14: number of entries in the address table
+            int addressEntryCount = exportTableHeader.ReadInt32();
+            // +0x18: number of name pointers
+            int namePointerCount = exportTableHeader.ReadInt32();
+            // +0x1C: export address table RVA
+            int addressTableRVA = exportTableHeader.ReadInt32();
+            // +0x20: name pointer RVA
+            int namePointerRVA = exportTableHeader.ReadInt32();
+            // +0x24: ordinal table RVA
+            int ordinalTableRVA = exportTableHeader.ReadInt32();
+
+            int[] addressTable = new int[addressEntryCount];
+            BlobReader addressTableReader = peImage.GetReader(peReader.GetOffset(addressTableRVA), sizeof(int) * addressEntryCount);
+            for (int entryIndex = 0; entryIndex < addressEntryCount; entryIndex++)
+            {
+                addressTable[entryIndex] = addressTableReader.ReadInt32();
+            }
+
+            ushort[] ordinalTable = new ushort[namePointerCount];
+            BlobReader ordinalTableReader = peImage.GetReader(peReader.GetOffset(ordinalTableRVA), sizeof(ushort) * namePointerCount);
+            for (int entryIndex = 0; entryIndex < namePointerCount; entryIndex++)
+            {
+                ushort ordinalIndex = ordinalTableReader.ReadUInt16();
+                ordinalTable[entryIndex] = ordinalIndex;
+                _ordinalRva.Add(entryIndex + minOrdinal, addressTable[ordinalIndex]);
+            }
+
+            BlobReader namePointerReader = peImage.GetReader(peReader.GetOffset(namePointerRVA), sizeof(int) * namePointerCount);
+            for (int entryIndex = 0; entryIndex < namePointerCount; entryIndex++)
+            {
+                int nameRVA = namePointerReader.ReadInt32();
+                if (nameRVA != 0)
+                {
+                    int nameOffset = peReader.GetOffset(nameRVA);
+                    BlobReader nameReader = peImage.GetReader(nameOffset, peImage.Length - nameOffset);
+                    StringBuilder nameBuilder = new StringBuilder();
+                    for (byte ascii; (ascii = nameReader.ReadByte()) != 0;)
+                    {
+                        nameBuilder.Append((char)ascii);
+                    }
+                    _namedExportRva.Add(nameBuilder.ToString(), addressTable[ordinalTable[entryIndex]]);
+                }
+            }
+        }
+
+        public static PEExportTable Parse(PEReader peReader)
+        {
+            return new PEExportTable(peReader);
+        }
+
+        public bool TryGetValue(string exportName, out int rva) => _namedExportRva.TryGetValue(exportName, out rva);
+        public bool TryGetValue(int ordinal, out int rva) => _ordinalRva.TryGetValue(ordinal, out rva);
```

Comment:
>TryGetValue [](start = 20, length = 11)

Is this overload being used? If not, consider removing it and `_ordinalRva`.

Reply:
This was an easy case because it's just a delete. I'm otherwise reluctant to do surgery here because this is mostly copying from the runtime repository. It's more "code we use" than "code we are super qualified to change" :smile: